File: Login/app/src/main/java/com/example/login/RegistroActivity.java
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class RegistroActivity extends AppCompatActivity {

    private SQLiteDatabase database;
    private EditText editTextNombre, editTextCorreo, editTextContrasena, editTextTipoUsuario;
    private Button buttonRegistrar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        DatabaseHelper dbHelper = new DatabaseHelper(this);
        dbHelper.copyDatabase();

        // Abre la base de datos después de copiarla desde assets
        database = dbHelper.getWritableDatabase();

        editTextNombre = findViewById(R.id.editTextNombre);
        editTextCorreo = findViewById(R.id.editTextCorreo);
        editTextContrasena = findViewById(R.id.editTextContrasena);
        editTextTipoUsuario = findViewById(R.id.editTextTipoUsuario);
        buttonRegistrar = findViewById(R.id.buttonRegistrar);

        buttonRegistrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                registrarUsuario();
            }
        });
    }

    private void registrarUsuario() {
        String nombre = editTextNombre.getText().toString().trim();
        String correo = editTextCorreo.getText().toString().trim();
        String contrasena = editTextContrasena.getText().toString().trim();
        String tipoUsuario = editTextTipoUsuario.getText().toString().trim();

        // Resto del código para registrar un nuevo usuario en la base de datos
        // ...

        // Ejemplo de cómo mostrar un mensaje de éxito
        Toast.makeText(this, "Registro exitoso", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onDestroy() {
        // Cierra la base de datos al destruir la actividad
        if (database != null) {
            database.close();
        }

        super.onDestroy();
    }
}
}

File: KotlinPrueba/app/src/main/java/com/example/kotlinprueba/RegistroActivity.kt
package com.example.kotlinprueba

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.FirebaseDatabase

class RegistroActivity : AppCompatActivity() {

    private lateinit var editTextUsername: EditText
    private lateinit var editTextPassword: EditText
    private lateinit var editTextEmail: EditText
    private lateinit var editTextBirthdate: EditText
    private lateinit var buttonRegister: Button
    private lateinit var auth: FirebaseAuth
    private lateinit var database: FirebaseDatabase

    private val TAG = "RegistroActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registro)

        editTextUsername = findViewById(R.id.editTextUsername)
        editTextPassword = findViewById(R.id.editTextPassword)
        editTextEmail = findViewById(R.id.editTextEmail)
        editTextBirthdate = findViewById(R.id.editTextBirthdate)
        buttonRegister = findViewById(R.id.buttonRegister)

        auth = FirebaseAuth.getInstance()
        database = FirebaseDatabase.getInstance()

        buttonRegister.setOnClickListener {
            registerUser()
        }
    }

    private fun registerUser() {
        val username = editTextUsername.text.toString()
        val password = editTextPassword.text.toString()
        val email = editTextEmail.text.toString()
        val birthdate = editTextBirthdate.text.toString()

        if (username.isNotEmpty() && password.isNotEmpty() && email.isNotEmpty() && birthdate.isNotEmpty()) {
            auth.createUserWithEmailAndPassword(email, password).addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    Log.d(TAG, "Usuario autenticado con éxito")
                    val user = auth.currentUser
                    val userRef = database.reference.child("usuarios").push()
                    val userId = userRef.key

                    val userMap = hashMapOf(
                        "id" to userId,
                        "n_usuario" to username,
                        "contrasena" to password,
                        "email" to email,
                        "fecha_nacimiento" to birthdate
                    )

                    if (userId != null) {
                        userRef.setValue(userMap).addOnCompleteListener { dbTask ->
                            if (dbTask.isSuccessful) {
                                Log.d(TAG, "Usuario registrado con éxito: $userMap")
                                Toast.makeText(this, "Usuario registrado con éxito", Toast.LENGTH_SHORT).show()
                            } else {
                                Log.e(TAG, "Error al registrar usuario en la base de datos: ${dbTask.exception?.message}")
                                Toast.makeText(this, "Error al registrar usuario", Toast.LENGTH_SHORT).show()
                            }
                        }
                    } else {
                        Log.e(TAG, "Error al obtener el ID del usuario")
                        Toast.makeText(this, "Error al registrar usuario", Toast.LENGTH_SHORT).show()
                    }
                } else {
                    Log.e(TAG, "Error al registrar usuario: ${task.exception?.message}")
                    Toast.makeText(this, "Error al registrar usuario: ${task.exception?.message}", Toast.LENGTH_SHORT).show()
                }
            }
        } else {
            Log.w(TAG, "Por favor, complete todos los campos")
            Toast.makeText(this, "Por favor, complete todos los campos", Toast.LENGTH_SHORT).show()
        }
    }
}
